homeassistant/components/influxdb.py

```diff
@@ -220,9 +221,11 @@ def event_to_json(event):
                         json['fields'][key] = float(
                             RE_DECIMAL.sub('', new_value))
 
-                # Infinity is not a valid float in InfluxDB
-                if (key, float("inf")) in json['fields'].items():
-                    del json['fields'][key]
+                # Infinity and NaN are not valid floats in InfluxDB
+                if key in json['fields']:
+                    converted = json['fields'][key]
+                    if math.isinf(converted) or math.isnan(converted):
```

Comment:
Not sure if a problem here, but `math.is...` will raise ValueErrors if not passes floats (for example strings) which was not the previous behavior.

Looking at the code, this would probably only be triggered if an attribute is called `something_str` but still it might be good to have a check in place if there are other edge cases.

Reply:
It seems to be TypeError, not ValueError(?). I now added a check for that but I don't think it is actually possible to hit, so no test.